src/tests/Microsoft.PowerFx.Interpreter.Tests/MiscInterpreterTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.PowerFx.Core.Tests;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Public.Values;
+using Xunit;
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class MiscInterpreterTests
```

Comment:
>MiscInterpreterTests [](http://example.com/codeflow?start=17&length=20)

Can you add a .txt test case here too (or even instead of the C# ones)? 

[https://github.com/microsoft/Power-Fx/tree/main/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases](https://github.com/microsoft/Power-Fx/tree/main/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases)

They are portable and easy to  write (and hence hit lots of corner cases) :)

RandBetween(5,1) // error case, wrong 
RandBetween(100.5,100.7) ? non integer values

You could also do looping with ForAll+Sequence:

Sum(ForAll(Sequence(10), RandBetween(1,5)), Value)

Reply:
Sure, no problem, I understand the first test in that it's looking for an error case. For the second one because in PowerFx a NumberValue is not an integer, it will evaluate that as the same as RandomBetween(100,100) and return 100. I'm not sure how PowerFx deals with evaluating parameters to be integers if the NumberValue passed in is based on a double, unless the RandomBeween functions needs to do manual validation in the function itself, though I didn't see this anywhere else though I may have just missed it. I can do this, just wasn't sure the intent in PowerFx where parameters are provided as NumberValue and they need to be integers.

I'm also not sure how to evaluate the result of something like Sum(ForAll(Sequence(10), RandBetween(1,5)), Value) in a .txt file test case as the result will be different based on the sum of the random values. I'm also going to leave the C# tests as they are checking the positive result or at least a set of values to see if they are in an expected range which I'm not sure how you can do with a .txt file based test if you can't put a function into the expected result that can evaluate something other than exact result values.

Reply:
You could probably do something like 

```
>> Sum(ForAll(Sequence(10), With({RandomNumber: RandBetween(1,20)}, If(RandomNumber >1 && RandomNumber < 20, 1, 0))), Value)
10
```

Reply:
@lesaltzm That makes sense, evaluate each iteration and just count that we got to the final expected count.

Reply:
Actually, this can be simpler
```
>> Sum(Sequence(100), With({RandomNumber: RandBetween(1,20)}, If(RandomNumber >1 && RandomNumber < 20, 1, 0))))
100

Reply:
With that assistance I was able to re-write all the tests I had in C# to be expressions that can run within a .txt file and they will test as much as the C# tests did before. So I'll remove the coded C# tests.